### DISCUSSION TITLE
Select menu in multiple languages

### DISCUSSION THREAD

**Discussion by airanbertinetti** [forntoh/LcdMenu#34] [Ideas]
Your library is magnificent. I'm using it on a NODE MCU8266 and it works great using I2C display. I navigate the menu with a rotary encoder and the ClickEncoder.h library.
I also created an exclusive function that performs the adjustment of parameters, since with the imputItem() I was unsuccessful. Now with its update menu.hide() and menu.show() it is much better, as it made it easier to return to the previous menu.
I also make that all parameter values ​​that are changed in the menu are saved in the EEPROM/FLASH memory of the 8266.
My project needs the menu to be in 3 different languages ​​(English, Spanish and Portuguese) as I live in Brazil. But as the text definition of the Menu Items are pointers of character pointers (char*), but I can only put them in the code scope. Any changes I make later by changing the values ​​at the memory address indicated by the pointer do not change anything in the menu. Although the values ​​in the memory addresses have already been changed, the menu does not change their text.
Could you help me solve this?

**Comment by forntoh** [comment]
Okay, that's a new feature.

It could be made possible by using an array as the `text` of a `MenuItem`, while creating the `MenuItem`, you should add the text in different languages, making sure that the order is the same for all the `MenuItem`s.
Then adding a new `MenuItem` type that will permit you to cycle through the array `text` when `enter()` is called.

> This will increase RAM usage thereby limiting the RAM available for the program during runtime.
> But it should be no problem with boards of higher RAM capacity. The Arduino might struggle at this.

**Reply by airanbertinetti** [reply]
Thanks for answering me, your work is great, congratulations.! 
But in my project, I don't need the 3 languages ​​to be present simultaneously. When the device is turned on, the menu start in standard English language. Only then the user will choose which language the menu should be. I do this by writing in the EEPROM a flag 0, 1 or 2 that will be read at startup and will correspond to the language that was chosen. This choice will happen in a menuItem "Language".
Unfortunately I still don't know how to implement this!
